pkg/pod/pod_test.go

```diff
@@ -2006,6 +2006,147 @@ debug-fail-continue-heredoc-randomly-generated-mz4c7
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		}, {
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		}},
+	}, {
+		desc: "with sidecar resource requirements",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+			Sidecars: []v1beta1.Sidecar{{
+				Name: "sidecar",
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU: resource.MustParse("750m"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU: resource.MustParse("1.5"),
+					},
+				},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("2"),
+					corev1.ResourceMemory: resource.MustParse("2Gi"),
+				},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("2"),
+				corev1.ResourceMemory: resource.MustParse("2Gi"),
+			},
+		}, {
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU: resource.MustParse("750m"),
+			},
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU: resource.MustParse("1.5"),
+			},
+		}},
+	}}
+
+	for _, tc := range testcases {
+		t.Run(tc.desc, func(t *testing.T) {
+			names.TestingSeed()
+			store := config.NewStore(logtesting.TestLogger(t))
+			enableAlphaAPI := map[string]string{"enable-api-fields": "alpha"}
+			store.OnConfigChanged(
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{Name: config.GetFeatureFlagsConfigName(), Namespace: system.Namespace()},
+					Data:       enableAlphaAPI,
+				},
+			)
+
+			kubeclient := fakek8s.NewSimpleClientset(
+				&corev1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: "default", Namespace: "default"}},
+			)
+			builder := Builder{
+				Images:     images,
+				KubeClient: kubeclient,
+			}
+			tr := &v1beta1.TaskRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo-taskrun",
+					Namespace: "default",
+				},
+				Spec: tc.trs,
+			}
+
+			gotPod, err := builder.Build(store.ToContext(context.Background()), tr, tc.ts)
+			if err != nil {
+				t.Fatalf("builder.Build: %v", err)
+			}
+
+			if err := verifyTaskLevelComputeResources(tc.expectedComputeResources, gotPod.Spec.Containers); err != nil {
+				t.Errorf("verifyTaskLevelComputeResources: %v", err)
+			}
+		})
+	}
+}
+
+// verifyTaskLevelComputeResources verifies that the given TaskRun's containers have the expected compute resources.
+func verifyTaskLevelComputeResources(expectedComputeResources []corev1.ResourceRequirements, containers []corev1.Container) error {
```

Comment:
👍 could you please also check the container name? This will help us test that the sidecar is getting the right resource requirements, as opposed to the step containers